specs/Freeze.md

```diff
@@ -0,0 +1,155 @@
+to access this feature so that invisible WebView can use less resources. We'd appreciate your feedback.
+
+
+# Description
+You may call the `TryFeezeAsync` API to have the WebView2 consume less memory. This is useful when your Win32 app becomes invisible, or when your Universal Windows Platform app is suspended, during the suspended event handler before completing the suspended event.
+
+# Examples
+## .Net, WinRT
+```c#
+async protected void OnSuspending(object sender, SuspendingEventArgs args)
+{
+    SuspendingDeferral deferral = args.SuspendingOperation.GetDeferral();
+    // Set webView.Visibility to false first.
+    // WebView2 must be invisible for TryFreezeAsync to succeed.
+    webView.Visibility = false;
+    await webView.CoreWebView2.TryFreezeAsync();
+    deferral.Complete();
+}
+async protected void OnSuspending(object sender, Object args)
+{
+    // Making a WebView2 visible will automatically unfreeze it
+    // But you can also explicitly call Unfreeze without making a WebView2 visible to unfreeze it.
+    webView.CoreWebView2.Unfreeze();
+    webView.Visibility = true;
+}
+```
+## Win32 C++
+As unfreeze is very fast and automatically happens when WebView becomes visible, the app can generaly immediately call `TryFreeze` when a WebView becomes invisible.
+```cpp
+bool ViewComponent::HandleWindowMessage(
+    HWND hWnd, UINT message, WPARAM wParam, LPARAM lParam, LRESULT* result)
+{
+    if (message == WM_SYSCOMMAND)
+    {
+        if (wParam == SC_MINIMIZE)
+        {
+            // Hide the webview when the app window is minimized, and freeze it.
+            m_controller->put_IsVisible(FALSE);
+            Freeze();
+        }
+        else if (wParam == SC_RESTORE)
+        {
+            // When the app window is restored, show the webview
+            // (unless the user has toggle visibility off).
+            if (m_isVisible)
+            {
+                Unfreeze();
+                m_controller->put_IsVisible(TRUE);
+            }
+        }
+    }
+}
+
+void ViewComponent::Freeze()
+{
+    HRESULT hr = webView->TryFreeze(
+        Callback<ICoreWebView2StagingTryFreezeCompletedHandler>(
+            [](HRESULT errorCode, BOOL isSuccessful) -> HRESULT {
+                std::wstringstream formattedMessage;
+                formattedMessage << "TryFreeze result (0x" << std::hex << errorCode
+                                 << ") " << (isSuccessful ? "succeeded" : "failed");
+                MessageBox(nullptr, formattedMessage.str().c_str(), nullptr, MB_OK);
+                return S_OK;
+            })
+            .Get());
+    if (FAILED(hr))
+        ShowFailure(hr, L"Call to TryFreeze failed");
+}
+
+void ViewComponent::Unfreeze()
+{
+    wil::com_ptr<ICoreWebView2Staging2> webView;
+    webView = m_webView.query<ICoreWebView2Staging2>();
+    webView->Unfreeze();
+}
+```
+
+# Remarks
+The CoreWebView2Controller must be invisible when the API is called. Otherwise, the
```

Comment:
More clearly, the CoreWebView2Controller.IsVisible must be false?

Reply:
updated